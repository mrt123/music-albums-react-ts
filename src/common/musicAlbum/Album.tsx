import AddToCartButton from "../AddToCartButton";
import AlbumDescription, { AlbumDescriptionProps } from "./AlbumDescription";
import AlbumImage from "./AlbumImage";
import AlbumNumber from "./AlbumNumber";

const albumStyle = {
  display: "flex",
  alignItems: "center",
  // justifyContent: "space-between",
  margin: "30px",
  fontSize: "12px",
  background: "#353a45",
  padding: "10px",
};

export interface AlbumProps {
  number: number;
  coverImageUrl: string;
  description: AlbumDescriptionProps;
}

const Album = ({ number, coverImageUrl, description }: AlbumProps) => {
  return (
    <div style={albumStyle}>
      <AlbumNumber value={number} />
      <AlbumImage imageUrl={coverImageUrl} />
      <AlbumDescription
        title={description.title}
        artist={description.artist}
        price={description.price}
      />
      <AddToCartButton />
    </div>
  );
};

export default Album;
