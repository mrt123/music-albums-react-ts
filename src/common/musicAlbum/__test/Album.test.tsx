import { render } from "@testing-library/react";
import Album from "../Album";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

const mockStore = configureStore();

const initialState = {
  shoppingCart: {
    items: [],
  },
};

describe("Album.tsx", () => {
  it("matches snapshot", () => {
    const fakeStore = mockStore(initialState);

    const imageDescription = {
      title: "MÄ™ka przez testy",
      artist: "DJ ZDFRONPol11",
      price: 7,
    };

    const { container } = render(
      <Provider store={fakeStore}>
        <Album
          id="1"
          number={1}
          coverImageUrl="test-image-url"
          description={imageDescription}
        />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
