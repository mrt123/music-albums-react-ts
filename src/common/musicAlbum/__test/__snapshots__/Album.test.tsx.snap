// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album.tsx matches snapshot 1`] = `
<div>
  <div
    style="display: flex; align-items: center; justify-content: space-around; margin: 30px; font-size: 12px; background: rgb(53, 58, 69); padding: 10px 20px 10px 10px;"
  >
    <div
      style="display: flex; align-items: center; padding: 5px; margin: 30px; font-size: 30px; font-family: 'Brush Script MT', cursive; color: rgb(237, 92, 145);"
    >
      1
    </div>
    <div
      style="position: relative;"
    >
      <div
        style="width: 100px; min-width: 100px; height: 100px; background: blue; background-image: url(test-image-url); background-size: contain; border-radius: 50%; margin: 10px;"
      />
      <div
        style="position: absolute; top: 0px; left: 0px;"
      />
    </div>
    <div
      style="margin: 8px;"
    >
      <div
        style="color: rgba(216, 237, 92, 0.612); font-weight: bold; font-size: 18px; font-family: 'Courier New', monospace; width: 330px; text-overflow: ellipsis; overflow: hidden; white-space: nowrap;"
      >
        MÄ™ka przez testy
      </div>
      <div
        style="margin: 10px; font-weight: 200; font-size: 14px; width: 330px;"
      >
        DJ ZDFRONPol11
      </div>
      <div
        style="margin: 10px; font-weight: 200; font-size: 14px; width: 330px;"
      >
        7
      </div>
    </div>
    <button
      data-testid="addToCartButton"
      style="background: rgb(237, 92, 145); cursor: pointer; border-radius: 4px; padding: 12px;"
    >
      Add
    </button>
  </div>
</div>
`;
